### PR TITLE
Add javadoc comments for gltfio.

### PR DESCRIPTION
**Pull request by prideout**

### CODE REVIEW

File: android/gltfio-android/src/main/java/com/google/android/filament/gltfio/AssetLoader.java [hunk 1]
```diff
@@ -38,6 +74,7 @@
     static Constructor<IndirectLight> sIndirectLightConstructor;
     static Constructor<Skybox> sSkyboxConstructor;
 
+    /** Initializes the gltfio JNI layer. Must be called before using any gltfio functionality. */
```
Comment (romainguy) [inline android/gltfio-android/src/main/java/com/google/android/filament/gltfio/AssetLoader.java]: Nit, but the typical Javadoc style is to write:

```
/**
 * Single line doc.
 */
```

It helps clearly see javadoc comments in the file.

File: android/gltfio-android/src/main/java/com/google/android/filament/gltfio/AssetLoader.java [hunk 2]
```diff
@@ -60,6 +97,11 @@ public static void init() {
         }
     }
 
+    /**
```
Comment (romainguy) [inline android/gltfio-android/src/main/java/com/google/android/filament/gltfio/AssetLoader.java]: You should add a line describing what the constructor does.

File: android/gltfio-android/src/main/java/com/google/android/filament/gltfio/AssetLoader.java [hunk 3]
```diff
@@ -75,21 +117,25 @@ public AssetLoader(@NonNull Engine engine, @NonNull MaterialProvider generator,
         }
     }
 
+    /** Frees all memory consumed by the native AssetLoader and its material cache. */
     public void destroy() {
         nDestroyAssetLoader(mNativeObject);
         mNativeObject = 0;
     }
 
+    /** Creates a FilamentAsset from the contents of a GLB file. */
```
Comment (romainguy) [inline android/gltfio-android/src/main/java/com/google/android/filament/gltfio/AssetLoader.java]: Link to other classes with `{@link FilamentAsset}` or at least use `<code>FilamentAsset</code>` to distinguish APIs from regular text.

File: android/gltfio-android/src/main/java/com/google/android/filament/gltfio/FilamentAsset.java [hunk 4]
```diff
@@ -21,6 +21,22 @@
 import com.google.android.filament.Box;
 import com.google.android.filament.Entity;
 
+/**
+ * Owns a bundle of Filament objects that have been created by AssetLoader.
```
Comment (romainguy) [inline android/gltfio-android/src/main/java/com/google/android/filament/gltfio/FilamentAsset.java]: Use @link as explained above for all classes you reference

File: android/gltfio-android/src/main/java/com/google/android/filament/gltfio/FilamentAsset.java [hunk 5]
```diff
@@ -21,6 +21,22 @@
...
+ * a TransformManager component, and some entities also have a NameComponentManager and/or
+ * RenderableManager components.
+ *
+ * In addition to the aforementioned entities, an asset has strong ownership over a list of
+ * VertexBuffer, IndexBuffer, MaterialInstance, Texture, and, optionally, a simple animation engine
+ * (Animator).
+ *
+ * Clients can ResourceLoader to create Texture objects, compute tangent quaternions, and upload
+ * data into vertex buffers and index buffers.
+ */
```
Comment (romainguy) [inline android/gltfio-android/src/main/java/com/google/android/filament/gltfio/FilamentAsset.java]: It is good practice to use @see at the end to point to related classes, APIs, etc.

```
/**
 * This class uses a {@link Foo} to do bar.
 * @see Foo
 * @see Foo#bar()
 */
```

**Comment by prideout** [comment]
Thanks Romain, that was good feedback.  Added a commit.

**Pull request merged**